src/shortest_path/average_length.rs

```diff
@@ -61,15 +63,18 @@ pub fn compute_distance_sum<Ty: EdgeType + Sync>(
             }
             level += 1
         }
-        count
+        (count, conn_pairs - 1)
     };
     let node_indices: Vec<NodeIndex> = graph.node_indices().collect();
     if n < parallel_threshold {
-        node_indices.iter().map(|index| bfs_traversal(*index)).sum()
+        node_indices
+            .iter()
+            .map(|index| bfs_traversal(*index))
+            .fold((0, 0), |a, b| (a.0 + b.0, a.1 + b.1))
     } else {
         node_indices
             .par_iter()
             .map(|index| bfs_traversal(*index))
-            .sum()
+            .reduce(|| (0, 0), |a, b| (a.0 + b.0, a.1 + b.1))
```

Comment:
Is there a reason you used `reduce()` here vs `fold()` in the serial case?

Reply:
Yeah, parallel `fold` produces a group of values rather than  a single output value.